tests/mockboto.py

```diff
@@ -31,28 +30,58 @@
 except ImportError:
     boto = None
 
+from mrjob.compat import version_gte
 from mrjob.conf import combine_values
 from mrjob.parse import is_s3_uri
 from mrjob.parse import parse_s3_uri
 from mrjob.parse import RFC1123
 
-DEFAULT_MAX_JOB_FLOWS_RETURNED = 500
-DEFAULT_MAX_DAYS_AGO = 61
+# list_clusters() only returns this many results at a time
+DEFAULT_MAX_CLUSTERS_RETURNED = 50
 
 # Size of each chunk returned by the MockKey iterator
 SIMULATED_BUFFER_SIZE = 256
 
-# versions of hadoop available on each AMI version. The EMR API treats None
-# and "latest" as separate logical AMIs, even though they're actually the
-# same AMIs as 1.0 and whatever they most recently released.
-AMI_VERSION_TO_HADOOP_VERSIONS = {
-    None: ['0.18', '0.20'],
-    '1.0': ['0.18', '0.20'],
-    '2.0': ['0.20.205'],
-    '2.0.0': ['0.20.205'],
-    'latest': ['0.20.205'],
+# what partial versions and "latest" map to, as of 2015-07-15
+AMI_VERSION_ALIASES = {
+    None: '1.0.0',  # API does this for old accounts
+    '2.0': '2.0.6',
+    '2.1': '2.1.4',
+    '2.2': '2.2.4',
+    '2.3': '2.3.6',
+    '2.4': '2.4.11',
+    '3.0': '3.0.4',
+    '3.1': '3.1.4',
+    '3.2': '3.2.3',
+    '3.3': '3.3.2',
+    '3.4': '3.4.0',
+    '3.5': '3.5.0',
+    '3.6': '3.6.0',
+    '3.7': '3.7.0',
+    '3.8': '3.8.0',
+    'latest': '2.4.2',
```

Comment:
Yes. Apparently no one actually wanted to have their production AMI suddenly change out from under them, and people who were using `latest` complained (why these people didn't just pick a particular AMI version, I don't know).

I think nowadays, (for non-ancient accounts) the API automatically chooses the (actual) latest AMI unless you explicitly specify one. In v0.5.x mrjob picks a particular default AMI for each version, just to be safe.